enhancements/bundle-validation.md

```diff
@@ -0,0 +1,221 @@
+    - E09: The packageName in bundle does not match repository name
+    - E10: duplicate CSV in bundle set
+    - E11: error getting spec.replaces from bundle CSV
+    - E12: spec.replaces is referenced by more than one CSV
+    - E13: spec.replaces field cannot match its own metadata.name
+    - E15: spec.replaces CSV is not present in manifests
+    - E16: spec.skips cannot match spec.replaces
+    - E17: metadata.annotations.skipPackageAnnotationKey range contains the current CSV's version
+    - E18: metadata.annotations.skipPackageAnnotationKey is an invalid semantic version range
+    - E19: spec.replaces references a parent in CSV replace chain which causes repalce cycle
+    - E20: spec.replaces does not map to any CSV in bundles
+    - E21: currentCSV for channel name not found in bundle
+
+- Validating cluster service versions for operatorhub.io UI **\***
+    - W01: CRD does not have an example CR in alm-examples
+    - W02: every owned CRD should have alm-examples
+    - E01: csv.spec.provider should be a singleton list
+    - E02: csv.spec.provider element should contain name as a single field
+    - E03: csv.spec.maintainers element should contain both name and email
+    - E04: csv.spec.maintainers element email is not valid
+    - E05: csv.spec.maintainers must be a list of name & email pairs
+    - E06: csv.spec.links element should contain both name and url
+    - E07: csv.spec.links element url is not valid
+    - E08: csv.spec.links must be a list of name & url pairs
+    - E09: spec.version is not a valid semver
+    - E10: metadata.annotations.capabilities is not a valid capabilities level if it is not ["Basic Install", "Seamless Upgrades", "Full Lifecycle", "Deep Insights", "Auto Pilot"]
+    - E11: spec.icon.mediatype %s is not a valid mediatype if it is not [\"image/gif\", \"image/jpeg\", \"image/png\", \"image/svg+xml\"]
+    - E12: spec.icon can only contain two fields: \"base64data\" and \"mediatype\"
+    - E13: spec.icon should be a singleton list
+    - E14: category is not valid if it is not [
+                "AI/Machine Learning",
+                "Application Runtime",
+                "Big Data",
+                "Cloud Provider",
+                "Developer Tools",
+                "Database",
+                "Integration & Delivery",
+                "Logging & Tracing",
+                "Monitoring",
+                "Networking",
+                "OpenShift Optional",
+                "Security",
+                "Storage",
+                "Streaming & Messaging"
+            ]
+
+**\*** Validation may be subjected to relocation or included as part of extended package.
+
+### Risks and Mitigations
+
+- If any set of rules is outdated or ill-formed that reports false-negative alerts, valid Operators may be blocked from being released or installed. Since we try to be conservative in reporting all possible errors, we might let some invalid configurations slide.
```

Comment:
We should explicitly allow users to disable specific checks so that we can always work around temporary problems.